broker/cluster.go

```diff
@@ -0,0 +1,170 @@
+package broker
+
+import (
+	"log"
+	"time"
+
+	"github.com/Percona-Lab/percona-dbaas-cli/dbaas"
+	"github.com/Percona-Lab/percona-dbaas-cli/dbaas/psmdb"
+	"github.com/Percona-Lab/percona-dbaas-cli/dbaas/pxc"
+)
+
+const (
+	defaultVersion = "default"
+
+	noS3backupWarn = `[Error] S3 backup storage options doesn't set: %v. You have specify S3 storage in order to make backups.
+You can skip this step by using --s3-skip-storage flag add the storage later with the "add-storage" command.
+`
+)
+
+func (p *Controller) DeployCluster(instance ServiceInstance, skipS3Storage *bool, instanceID string) error {
+	switch instance.ServiceID {
+	case pxcServiceID:
+		app := pxc.New(instance.Parameters.ClusterName, defaultVersion)
+		conf := dbaas.ClusterConfig{}
+		SetDefault(&conf)
+
+		var s3stor *dbaas.BackupStorageSpec
+
+		setupmsg, err := app.Setup(conf, s3stor)
+		if err != nil {
+			log.Println("[Error] set configuration:", err)
+			return nil
+		}
+
+		log.Println(setupmsg)
+
+		created := make(chan string)
+		msg := make(chan dbaas.OutuputMsg)
+		cerr := make(chan error)
+
+		go dbaas.Create("pxc", app, created, msg, cerr)
+
+		for {
```

Comment:
All this workaround is needed for the CLI to have a user interaction (show spinner/progress, errors/messages, etc.) while the payload is executing.